### PR TITLE
Migrate tasks 6-10 from MarkingOracles Kata

### PR DESCRIPTION
**Pull request by jkingdon-ms**
This resolves part of https://github.com/microsoft/qsharp/issues/1185 .

### CODE REVIEW

File: katas/content/marking_oracles/index.md [hunk 1]
```diff
@@ -101,6 +101,51 @@ This kata continues the exploration of quantum oracles started in the Oracles ka
     ]
 })
 
+@[exercise]({
+    "id": "marking_oracles__contains_substring",
+    "title": "Contains Substring",
```
Comment (tcNickolas) [inline katas/content/marking_oracles/index.md:106]:
```suggestion
    "title": "Does Bit String Contain Substring?",
```

File: katas/content/marking_oracles/index.md [hunk 2]
```diff
@@ -101,6 +101,51 @@ This kata continues the exploration of quantum oracles started in the Oracles ka
...
+    "path": "./contains_substring/",
+    "qsDependencies": [
+        "../KatasLibrary.qs"
+    ]
+})
+
+@[exercise]({
+    "id": "marking_oracles__balanced",
+    "title": "0s and 1s Balanced",
```
Comment (tcNickolas) [inline katas/content/marking_oracles/index.md:115]:
```suggestion
    "title": "Is Bit String Balanced?",
```

File: katas/content/marking_oracles/index.md [hunk 3]
```diff
@@ -101,6 +101,51 @@ This kata continues the exploration of quantum oracles started in the Oracles ka
...
+    "path": "./balanced/",
+    "qsDependencies": [
+        "../KatasLibrary.qs"
+    ]
+})
+
+@[exercise]({
+    "id": "marking_oracles__majority",
+    "title": "0s or 1s Majority",
```
Comment (tcNickolas) [inline katas/content/marking_oracles/index.md:124]:
```suggestion
    "title": "Majority Function",
```

File: katas/content/marking_oracles/index.md [hunk 4]
```diff
@@ -101,6 +101,51 @@ This kata continues the exploration of quantum oracles started in the Oracles ka
...
+    "title": "Bit Sum Divisible By 3",
+    "path": "./bit_sum_div_3/",
+    "qsDependencies": [
+        "../KatasLibrary.qs"
+    ]
+})
+
+@[exercise]({
+    "id": "marking_oracles__num_div_3",
+    "title": "Number Divisible By 3",
```
Comment (tcNickolas) [inline katas/content/marking_oracles/index.md:142]:
```suggestion
    "title": "Is Number Divisible by 3?",
```

**Review by tcNickolas — CHANGES_REQUESTED** [review]
Looks great! I left some comments, mostly on rearranging the code so that all operations are in the most logical places for them, and one on majority task.

Thank you!

File: katas/content/marking_oracles/majority/Solution.qs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+            for q in x {
+                Controlled IncrementBE([q], inc);
+            }
+        } apply {
+            CNOT(inc[log-1], y);
+            if log > 2 {
+                CCNOT(inc[0], inc[1], y);
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jkingdon-ms) [reply katas/content/marking_oracles/majority/Solution.qs:19]: i see, it is unfortunate we must duplicate so much code, but i suppose there is not easy way around it.

File: katas/content/marking_oracles/majority/Verification.qs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    open Microsoft.Quantum.Arrays;
+
+    function F_Majority(args : Bool[]) : Bool {
+        let N = Length(args);
+        return Count(x -> x, args) > (N - 1) / 2;
+    }
+
+    @EntryPoint()
+    operation CheckSolution() : Bool {
+        for n in [3, 5] {
```
[the remark this answers is not in the document]
Comment (jkingdon-ms) [reply katas/content/marking_oracles/majority/Verification.qs:12]: will add 7 and ensure solution passes for it as well

File: katas/content/marking_oracles/num_div_3/Solution.qs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+namespace Kata {
+    open Microsoft.Quantum.Katas;    
+    operation Oracle_DivisibleBy3 (x : Qubit[], y : Qubit) : Unit is Adj + Ctl {
+        use counter = Qubit[2];
+        within {
+            for i in 0 .. Length(x) - 1 { // Iterate starting from the least significant bit
```
[the remark this answers is not in the document]
Comment (jkingdon-ms) [reply katas/content/marking_oracles/num_div_3/Solution.qs:6]: ah this files comments slipped by me! good catch

**Review by tcNickolas — APPROVED** [review]
Looks great! Thank you so much for working on this!

**Pull request merged**